src/compile/common.ts

```diff
@@ -18,17 +18,18 @@ import {RepeaterValue, RepeatModel} from './repeat';
 import {UnitModel} from './unit';
 
 
-export function buildModel(spec: Spec, parent: Model, parentGivenName: string, repeater: RepeaterValue, config: Config): Model {
+export function buildModel(spec: Spec, parent: Model, parentGivenName: string,
```

Comment:
We can't do that.    For ordinal, if you explicitly set `"width"`, then you get flexible band scale.
If you don't set `"width"`, config.cell.width will be ignored. 

If you want to change this rule, propose better idea in another issue.  That's beyond the scope of this PR.